state-chain/cf-integration-tests/src/fee_scaling.rs

```diff
@@ -0,0 +1,143 @@
+use cf_primitives::{AccountRole, Asset, AuthorityCount, FLIPPERINOS_PER_FLIP};
+use cf_traits::{FeeScalingRateConfig, IncreaseOrDecrease};
+use frame_support::pallet_prelude::{InvalidTransaction, TransactionValidityError};
+use pallet_cf_flip::FeeScalingRate;
+use pallet_cf_pools::RangeOrderSize;
+use sp_keyring::Ed25519Keyring as AccountKeyring;
+use sp_runtime::FixedU64;
+use state_chain_runtime::{Flip, Runtime, RuntimeCall};
+
+use crate::network::apply_extrinsic_and_calculate_gas_fee;
+
+fn update_range_order_call(base_asset: Asset) -> RuntimeCall {
+	RuntimeCall::LiquidityPools(pallet_cf_pools::Call::update_range_order {
+		base_asset,
+		quote_asset: Asset::Usdc,
+		id: 0,
+		option_tick_range: Some(-10..10),
+		size_change: IncreaseOrDecrease::Decrease(RangeOrderSize::Liquidity {
+			liquidity: 1_000u128,
+		}),
+	})
+}
+
+#[test]
+fn fee_scales_within_a_pool() {
+	const EPOCH_BLOCKS: u32 = 100;
+	const MAX_AUTHORITIES: AuthorityCount = 10;
+	let lp = AccountKeyring::Alice;
+	super::genesis::with_test_defaults()
+		.epoch_duration(EPOCH_BLOCKS)
+		.max_authorities(MAX_AUTHORITIES)
+		.with_additional_accounts(&[(
+			lp.to_account_id(),
+			AccountRole::LiquidityProvider,
+			5 * FLIPPERINOS_PER_FLIP,
+		)])
+		.build()
+		.execute_with(|| {
+			let (mut testnet, _, _) =
+				crate::network::fund_authorities_and_join_auction(MAX_AUTHORITIES);
+
+			let lp_account_id = lp.to_account_id();
+
+			let call = update_range_order_call(Asset::Eth);
+
+			assert_eq!(FeeScalingRate::<Runtime>::get(), FeeScalingRateConfig::NoScaling);
+
+			let (mut last_gas, mut last_remaining_balance) =
+				apply_extrinsic_and_calculate_gas_fee(lp, call.clone()).unwrap();
+			(0..50).for_each(|_| {
+				let (gas, remaining_balance) =
+					apply_extrinsic_and_calculate_gas_fee(lp, call.clone()).unwrap();
+				assert_ne!(gas, 0);
+				assert_eq!(gas, last_gas);
+				last_gas = gas;
+				assert!(remaining_balance < last_remaining_balance);
+				last_remaining_balance = remaining_balance;
+			});
+
+			// Reset the fee scaling counters
+			testnet.move_forward_blocks(1);
+
+			// Set the config to scale per pool.
+			FeeScalingRate::<Runtime>::set(FeeScalingRateConfig::ExponentBuffer {
+				buffer: 0,
+				exp_base: FixedU64::from_rational(2, 1u128),
+			});
+
+			let mut can_pay = true;
+			(1..50).for_each(|_| {
+				match apply_extrinsic_and_calculate_gas_fee(lp, call.clone()) {
+					Ok((gas, remaining_balance)) => {
+						assert!(gas >= last_gas);
```

Comment:
If the fee is always the MAX (because it doesn't scale) then this test will always pass.

Reply:
(changing the check to `gas > last_gas` causes it to fail on the first iteration)